sycl/include/sycl/ext/intel/esimd/detail/memory_intrin.hpp

```diff
@@ -629,6 +640,12 @@ __esimd_gather_masked_scaled2(SurfIndAliasTy surf_ind, uint32_t global_offset,
 {
   static_assert(Scale == 0);
 
+  // determine the original element's type size (as __esimd_scatter_scaled
+  // requires vals to be a vector of 4-byte integers)
+  constexpr size_t OrigSize = __ESIMD_DNS::ElemsPerAddrDecoding(TySizeLog2);
+  using RestoredTy = std::conditional_t<sizeof(Ty) == OrigSize, Ty,
+                                        __ESIMD_DNS::uint_type_t<OrigSize>>;
```

Comment:
```suggestion
  using RestoredTy = __ESIMD_DNS::uint_type_t<OrigSize>>;
```

Reply:
@dongkyunahn-intel, can we commit this suggestion?
In addition to closing this comment, this should re-start the testing.